segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/StreamSegmentContainer.java

```diff
@@ -370,6 +379,61 @@ private CompletableFuture<Void> startWhenDurableLogOnline() {
         return isReady;
     }
 
+    /**
+     * Adjusting storage metadata segment length in container metadata.
+     * @return a CompletableFuture which when completed indicates successful updation
+     * of storage metadata length in container metadata.
+     */
+    private CompletableFuture<Void> adjustStorageMetadataLength() {
+        try {
+            // No-op for non-recovery container startup.
+            if (!shouldRecoverFromStorage().get()) {
+                log.info("{}: Non-recovery startup. No need to adjust storage metadata length", this.traceObjectId);
+                return CompletableFuture.completedFuture(null);
+            }
+            // Adjust always the storage metadata segment length in container metadata. This is the planned migration
+            // recovery usecase where we would flush-to-storage, so should be safe to invoke the below adjust-length flow if not taken effect.
+            ContainerTableExtension extension = this.getExtension(ContainerTableExtension.class);
```

Comment:
Over all looks good to me.
My only comment would be to use val instead of actual types. It not only makes code absolutely thread safe but also makes it more readable. (JVM memory model enforces cache coherency in case of final variables. Even if it is a really quite remote possibility there are thread visibility issues when non-final variables are passed to lambda that runs on a different thread.)

Reply:
using val